vultr/resource_instance.go

```diff
@@ -424,6 +424,12 @@ func resourceInstanceDelete(d *schema.ResourceData, meta interface{}) error {
 			if err.Error() == "Unable to destroy server: Unable to remove VM: Server is currently locked" {
 				continue
 			}
+			
+			// If server marked pending, break.
+			if err.Error() == "Unable to destroy server: Server is already pending destruction." {
+				break
```

Comment:
I think we should make this a `continue`. This way, Terraform will keep trying and wait until the server is fully destroyed before returning. Essentially, this is a similar case to the one above.